Src/FluentAssertions/Formatting/LineCollection.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Formatting;
+
+/// <summary>
+/// A collection of lines that will throw a <see cref="MaxLinesExceededException"/> when the number of lines
+/// exceeds the maximum.
+/// </summary>
+internal class LineCollection(int maxLines) : IEnumerable<Line>
+{
+    private readonly List<Line> lines = new();
+
+    public int Count => lines.Count;
+
+    public bool HasLinesBeyond(Line line)
+    {
+        // Null means that we're referring to the top of the list
+        return (line is null && lines.Count > 1) || (line is not null && lines.IndexOf(line) < (lines.Count - 1));
+    }
+
+    public void Add(Line line)
+    {
+        lines.Add(line);
+        OnCollectionIsModified();
+    }
+
+    public void AddLineAfter(Line line, Line newLine)
+    {
+        int index = lines.IndexOf(line);
+
+        Insert(index + 1, newLine);
+    }
+
+    public void InsertLineBefore(Line line, Line newLine)
+    {
+        int index = lines.IndexOf(line);
+
+        Insert(Math.Max(index, 0), newLine);
+    }
+
+    public void InsertAtTop(string fragment)
+    {
+        // If there's a single line, insert at the beginning of that line
+        // If there are more than one line, insert as a new line at the top
+        if (lines.Count == 1)
+        {
+            lines[0].InsertAtStart(fragment);
+        }
+        else
+        {
+            Insert(0, new Line(fragment));
+        }
```

Comment:
The mutation testing showed that this entire branch can be removed without affecting any tests.
I tried to verify this by setting a breakpoint in here, which wasn't hit when running tests.

Reply:
Same comment as before. I'm also very hesitant. Also, I took me a LOT of time trying to come up with a working solution, so if I remove those cases now and a scenario pops up that needs it again, it may take a lot of time to get there.

Reply:
This test (despite not a particular good one) exercises the else branch

```cs
[Fact]
public void Adding_a_fragment()
{
    // Arrange
    var formatter = new SomeFormatter1();
    using var _ = new FormatterScope(formatter);

    // Act
    string str = Formatter.ToString(null);

    // Assert
    str.Should().Be("fragment");
}

private class SomeFormatter1 : IValueFormatter
{
    public bool CanHandle(object value) => true;

    public void Format(object value, FormattedObjectGraph formattedGraph, FormattingContext context, FormatChild formatChild)
    {
        formattedGraph.GetAnchor().InsertLineOrFragment("fragment");
    }
}
```